services/sonar/sonar-redirector.service.js

```diff
@@ -4,15 +4,33 @@ const { redirector } = require('..')
 
 module.exports = [
   redirector({
+    name: 'SonarVersionPrefixRedirector',
     category: 'analysis',
     route: {
       base: 'sonar',
       pattern:
         ':sonarVersion/:protocol(http|https)/:host(.+)/:component(.+)/:metric',
     },
     transformPath: ({ protocol, host, component, metric }) =>
-      `/sonar/${protocol}/${host}/${component}/${metric}`,
-    transformQueryParams: ({ sonarVersion }) => ({ sonarVersion }),
-    dateAdded: new Date('2019-04-02'),
+      `/sonar/${component}/${metric}`,
+    transformQueryParams: ({ protocol, host, sonarVersion }) => ({
+      server: `${protocol}://${host}`,
+      sonarVersion,
+    }),
+    dateAdded: new Date('2019-07-05'),
+  }),
+  redirector({
+    name: 'SonarServerRedirector',
+    category: 'coverage',
+    route: {
+      base: 'sonar',
+      pattern: ':protocol(http|https)/:host(.+)/:component(.+)/:metric',
+    },
+    transformPath: ({ component, metric }) => `/sonar/${component}/${metric}`,
+    transformQueryParams: ({ protocol, host, sonarVersion }) => ({
```

Comment:
Do we need `sonarVersion` here? I believe this redirector will only be for the "current" routes (badges using the route created between when the Sonar service was refactored and now) and `sonarVersion` will already be specified as a query param.

I think this helper is only for converting route params to query params

Reply:
The current URL folks would be using for one of these Sonar badges with an older SonarQube version would be:
https://img.shields.io/sonar/http/sonar.petalslink.com/org.ow2.petals:petals-se-ase/coverage.svg?sonarVersion=4.2

(or in this PR review app)
https://shields-staging-pr-3696.herokuapp.com/sonar/http/sonar.petalslink.com/org.ow2.petals:petals-se-ase/coverage.svg?sonarVersion=4.2

I believe that's going to result in a redirect of:
(for the review app)
https://shields-staging-pr-3696.herokuapp.com/sonar/org.ow2.petals:petals-se-ase/coverage.svg?server=http%3A%2F%2Fsonar.petalslink.com

As the `sonarVersion` query param will be set to undefined by this redirect.  Note that particular badge will still render since the http://sonar.petalslink.com instance is Sonar 5.4 (which supports the new/modern API schema and the old/legacy API). However, I believe including `sonarVersion` here will break badges/users with legacy Sonar instances

Reply:
Ah, yes, I believe you're right!